### PR TITLE
Examples for README

### PR DESCRIPTION
**Pull request by sdiehl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,14 +7,43 @@ A IRBuilder, starting out as a thin reinterpretation of the C++ IRBuilder inside
...
+import qualified LLVM.AST.Constant as C
+
+import ModuleBuilder
+import IRBuilder.Monad
+import IRBuilder.Instruction
+
+simple :: IO ()
+simple = T.putStrLn $ ppllvm $ mkModule $ execModuleBuilder emptyModuleBuilder $ mdo
+
+  function "add" [(i32, Just "a"), (i32, (Just "b"))] i32 $ \[a,b] -> mdo
```
Comment (ollef) [inline README.md]: There's a superfluous paren around the second `Just` on this line.

**Comment by acowley** [comment]
I like the idea of a `Maybe`-like `ParameterName` type with an `IsString` instance!

**Comment by sdiehl** [comment]
Indeed, the ParameterName would need the instance added upstream in `llvm-hs.`

**Comment by ollef** [comment]
@sdiehl: Ah, is there an existing `ParameterName` type? I couldn't find such a thing in llvm-hs.

**Comment by cocreature** [comment]
> @sdiehl: Ah, is there an existing ParameterName type? I couldn't find such a thing in llvm-hs.

I don’t think we have such a type. I agree that adding something like that would be great! I’m generally quite fond of domain-specific types (assuming there is not a whole lot to be gained by reusing existing types) with descriptive constructor names so just adding this to `llvm-irbuilder` seems reasonable.

**Pull request merged**